quickwit/quickwit-ingest/src/ingest_v2/ingester.rs

```diff
@@ -961,6 +961,48 @@ impl Ingester {
 
         Ok(DecommissionResponse {})
     }
+
+    pub async fn mrecordlog_summary(&mut self) -> IngestV2Result<MRecordlogSummary> {
+        let rw_mrecordlog = self.state.mrecordlog();
+        // this is a debug api endpoint, with_lock_metrics! doesn't seem necessary
+        let maybe_mrecordlog = rw_mrecordlog.read().await;
+
+        let summary = match &*maybe_mrecordlog {
```

Comment:
Nitpick: let... else... are  nice for those.


```suggestion
        let Some(summary) = maybe_mrecordlog else {
               return Err(IngestV2Error::Internal(
                    "mrecordlog isn't initialized".to_string(),
                ))
        };
```

You could also use `.ok_or_else(|| {})?`